### ISSUE TITLE
getToken fails in authentication FrameContext

### ISSUE THREAD

**Issue by vhensorskyi** [OfficeDev/microsoft-teams-library-js#2107]
Hi,

We have been using teams-js library `2.6.0` and noticed that `await microsoftTeams.authentication.getAuthToken({ silent: true })` call fails with `Cannot get authorization url App webApplicationInfo or resource not defined in manifest` in `microsoftTeams.FrameContexts.authentication` but works fine in other contexts fine. Is it a known issue?

Regards,
Volodymyr

**Comment by AE-MS** [comment]
Thanks for reporting this! I tried to repro the issue but was successfully able to call `microsoftTeams.authentication.getAuthToken` and get a token from within the pop-up authentication window (which had a frame context of `authentication`.)

I am able to see the place in the code where the "App webApplicationInfo or resource not defined in manifest" string is defined, but it doesn't seem like it should be triggered in this scenario.

Could you provide some more details so I can investigate further and help?

1. Which host are you seeing this in? Outlook, Microsoft365 App, or Teams? If Teams, do you know if it's "New Teams" (aka V2) or "Legacy Teams" (aka V1)?
2. What platform are you seeing this? Web/Windows/Mac/iOS/Android?
3. I assume that your app manifest does actually define a resource in the `webApplicationInfo` section, but if you could confirm this, that would be great.
4. I assume that your app is triggering an authentication pop-up by calling `authentication.authenticate` and then calling `authentication.getAuthToken` from within that pop-up window (based on your comment about the frame context being `authentication`. Is this correct? What parameters is it passing when it calls `authentication.authenticate`?
5. Does calling `getAuthToken` from your app outside of an authentication pop-up (i.e. different frame context) work just fine?

Thanks!

**Comment by vhensorskyi** [comment]
Hello,

Thanks for your response.

1. I checked on Teams both versions as well as on Web and Mac Desktop clients and the issue exists. 

<img width="721" alt="Screenshot 2024-01-03 at 09 33 10" src="https://github.com/OfficeDev/microsoft-teams-library-js/assets/17333999/bd881fd1-9a96-41a0-9884-c081eb7493aa">

2. Web and Mac were used only

3. Yes, webApplicationInfo is set properly in the manifest
4. The is initiated from the card: 
> {
>                     "type": "Action.Submit",
>                     "title": "Sign In",
>                     "data": {
>                       "msteams": {
>                         "type": "signin",
>                         "value": "${authUrl}"
>                       }
>                     }
>                   }

 and the popup has authentication context.
5. Yes, using the same configuration `getAuthToken` fetches tokens fine in different contexts.

Let me know if any additional info is needed.

**Comment by AE-MS** [comment]
Thanks for the additional info. Triggering the authentication window from a card is likely the difference in how we are trying to reproduce the problem. Can you tell me more about how the card is being provided to Teams? E.g., is it being returned from a bot? Or a message extension?

**Comment by vhensorskyi** [comment]
Yes, it's an adaptive card shared by a bot in a personal conversation.

**Comment by AE-MS** [comment]
Thanks; with this information I was able to repro the problem. I will investigate and report back here.

**Comment by vhensorskyi** [comment]
Hi,

Is there any progress on this?

**Comment by AE-MS** [comment]
Thanks for checking in; I'm still working with the team responsible for the likely problematic code to identify the issue and understand their plan for the fix. I don't know of any workarounds, unfortunately.

**Comment by vhensorskyi** [comment]
Thanks for the info. Please keep me posted.

**Comment by AE-MS** [comment]
After reproducing the problem and debugging it based on your helpful information, it looks like the problem is in the Teams app itself (and not the application platform) so @nwojod-MSFT will transition the tracking of this issue to the folks who help with issues in Teams.

For reference for others: the issue only repros when the authentication pop-up is triggered from an adaptive card. It does not occur when the authentication pop-up is called from teamsjs using `authentication.authenticate`.

**Comment by vhensorskyi** [comment]
Thank you.
FYI, it also reproducible from the messaging extension cards :(

**Comment by Prasad-MSFT** [comment]
@vhensorskyi - We have raised a bug for the same. We will inform you once we get any further update from engineering team. Thanks!

**Comment by AE-MS** [comment]
Hi @vhensorskyi -- I just wanted to give you an update: I talked with the developer in Teams responsible for the code with the issue and they've coded up a fix for this and it's out for code review. I'll provide more updates as I have them.

**Comment by Prasad-MSFT** [comment]
Hi @vhensorskyi - We got update from engineering team that the fix has been merged. Please do verify and let us know.

**Comment by vhensorskyi** [comment]
Hi @Prasad-MSFT,

Thanks for getting back. I've just tested the flow again and it doesn't seem to be working properly. Now instead of `Cannot get authorization url App webApplicationInfo or resource not defined in manifest` the function throws `Error` without any details. Please see a screenshot for reference. It was verified on web teams v2.

<img width="802" alt="Screenshot 2024-06-13 at 11 04 01" src="https://github.com/OfficeDev/microsoft-teams-library-js/assets/17333999/1e4604f0-5d1f-4989-9541-39923fca1938">

**Comment by handicraftsman** [comment]
Hi @Prasad-MSFT ! Having the same issue, it is just `Error: Error` for us though.

It is failing in `authentication` context and in `getAuthToken` as well, as we can see in our Sentry dashboard.

**Comment by Prasad-MSFT** [comment]
Thanks for letting us know. We have informed the same to engineering team. Once we get any update, we will notify in this thread.

**Comment by vbatr** [comment]
Hi @Prasad-MSFT 

Having the same issue while trying to call auth flow from Messaging Extension (signInLink is calling getAuthToken()):

```
                return new MessagingExtensionResponse
                {
                    ComposeExtension = new MessagingExtensionResult
                    {
                        Type = "auth",
                        SuggestedActions = new MessagingExtensionSuggestedAction
                        {
                            Actions = new List<CardAction>
                                {
                                    new CardAction
                                    {
                                        Type = ActionTypes.OpenUrl,
                                        Value = signInLink,
                                        Title = "Service OAuth",
                                    },
                                },
                        },
                    },
                };
```

checked on "@microsoft/teams-js": "2.28.0"

**Comment by Prasad-MSFT** [comment]
@vhensorskyi, @handicraftsman, @vbatr - apologies for missing this thread. Kindly let us know if you are still experiencing this issue and if yes, please do share the relevant code snippets along with detailed issue description and platform details (Web or Desktop), so that we can investigate and raise bug if required. Thank you!

**Comment by VolodymyrHensorskyi** [comment]
Hi @Prasad-MSFT ,

Yes, the issue is still reproducible and the code snippet was shared as well. Let us know what other details are needed.

**Comment by vhensorskyi** [comment]
Hi @Prasad-MSFT,

Yes, the issue is still reproducible and the code snippet was shared as well. Let us know what other details are needed. Could you please reopen the issue?

**Comment by vbatr** [comment]
Hi @Prasad-MSFT 

yes, the issue is still relevant on all platforms - Web, Desktop, Mobile. The code snippets from the previous comments can be used for testing.

Regards,
Volodymyr B.
